crates/typst-library/src/meta/document.rs

```diff
@@ -59,7 +60,7 @@ impl LayoutRoot for DocumentElem {
 
             if let Some(page) = child.to::<PageElem>() {
                 let number = NonZeroUsize::ONE.saturating_add(pages.len());
-                let fragment = page.layout(vt, styles, number)?;
+                let fragment = page.layout(vt, styles, number, &mut prev_page_label)?;
```

Comment:
Similarly here, we need to keep track of the previous numbering if we want to avoid `PageLabel` repetition.
Though, I'd rather not modify this method like that but I can't see any other simple way that doesn't require iterating over something / introspection.

Reply:
okay